### ISSUE TITLE
Restoring parameters  hungs

### ISSUE THREAD

**Issue by zh794390558** [tensorflow/tensor2tensor#494]
I have the same problem with blstm,
with --schedule 'train'

>INFO:tensorflow:Global model_fn finished.
INFO:tensorflow:Create CheckpointSaverHook.
2017-12-29 07:38:44.128222: I tensorflow/core/platform/cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
2017-12-29 07:38:44.851032: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1030] Found device 0 with properties:
name: TITAN Xp major: 6 minor: 1 memoryClockRate(GHz): 1.582
pciBusID: 0000:05:00.0
totalMemory: 11.90GiB freeMemory: 11.74GiB
2017-12-29 07:38:44.851088: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: TITAN Xp, pci bus id: 0000:05:00.0, compute capability: 6.1)

, --schedule 'evalution'

>2017-12-29 07:43:23.631197: I tensorflow/core/platform/cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
2017-12-29 07:43:24.383548: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1030] Found device 0 with properties:
name: TITAN Xp major: 6 minor: 1 memoryClockRate(GHz): 1.582
pciBusID: 0000:05:00.0
totalMemory: 11.90GiB freeMemory: 11.74GiB
2017-12-29 07:43:24.383603: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: TITAN Xp, pci bus id: 0000:05:00.0, compute capability: 6.1)
INFO:tensorflow:Restoring parameters from d/model.ckpt-1

or '--schedule 'test''

>INFO:tensorflow:Create CheckpointSaverHook.
2017-12-29 07:52:49.675929: I tensorflow/core/platform/cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
2017-12-29 07:52:50.477620: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1030] Found device 0 with properties:
name: TITAN Xp major: 6 minor: 1 memoryClockRate(GHz): 1.582
pciBusID: 0000:05:00.0
totalMemory: 11.90GiB freeMemory: 11.74GiB
2017-12-29 07:52:50.477673: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: TITAN Xp, pci bus id: 0000:05:00.0, compute capability: 6.1)
INFO:tensorflow:Restoring parameters from d/model.ckpt-1

` --schedule 'continuous_train_and_eval'` seems work,

>totalMemory: 11.90GiB freeMemory: 11.74GiB
2017-12-29 08:15:39.798320: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: TITAN Xp, pci bus id: 0000:05:00.0, compute capability: 6.1)
INFO:tensorflow:Restoring parameters from d/model.ckpt-1
INFO:tensorflow:Saving checkpoints for 2 into d/model.ckpt.
INFO:tensorflow:loss = 5.00657, step = 2

but --schedule 'train_and_evaluate'

The problm seems like to https://github.com/tensorflow/tensor2tensor/issues/353, and i use `tf.nn.bidirectional_dynamic_rnn` without  'orthogonal initializer`

**Comment by zh794390558** [comment]
https://github.com/tensorflow/tensorflow/blob/r1.5/tensorflow/contrib/learn/python/learn/learn_runner.py#L272

```
  if config.task_type == run_config_lib.TaskType.MASTER:
    # TODO(rhaertel): handle the case where there is more than one master
    # or explicitly disallow such a case.
    return 'train_and_evaluate'
  elif config.task_type == run_config_lib.TaskType.PS:
    return 'run_std_server'
  elif config.task_type == run_config_lib.TaskType.WORKER:
    return 'train'

```
may be 'train_and_evaluate' is used for local trainning and run_config_lib.TaskType.MASTER: `train` is only for WORKER.

But I think this is caused by prams.

**Comment by rsepassi** [comment]
Please be more clear in which command hangs.

**Comment by rsepassi** [comment]
Closing for now. If you decide to reopen or rereport, please include T2T version, command lines, and stack trace.
